File: src/main/scala/scaled/code/PropertiesMode.scala
//
// Scaled Java Mode - a Scaled major mode for editing Java code
// http://github.com/scaled/java-mode/blob/master/LICENSE

package scaled.code

import scaled._
import scaled.grammar._
import scaled.major.CodeConfig

object PropertiesConfig extends Config.Defs {
  import CodeConfig._
  import GrammarCodeConfig._

  // map TextMate grammar scopes to Scaled style definitions
  val effacers = List(
    effacer("comment.line", commentStyle),
    effacer("keyword", keywordStyle)
  )
}

@Major(name="properties",
       tags=Array("code", "project", "properties"),
       pats=Array(".*\\.properties"),
       desc="A major mode for editing Java properties files.")
class PropertiesMode (env :Env) extends GrammarCodeMode(env) {

  override def configDefs = PropertiesConfig :: super.configDefs
  override protected def grammars = Grammars.propsGrammars
  override protected def effacers = PropertiesConfig.effacers
}

File: src/java/scaled/code/PropertiesMode.java
//
// Scaled Java Mode - support for editing Java code
// https://github.com/scaled/java-mode/blob/master/LICENSE

package scaled.code;

import scaled.*;
import scaled.grammar.GrammarCodeMode;

@Major(name="properties",
       tags={ "code", "project", "properties" },
       pats={ ".*\\.properties", "package.scaled", "module.scaled" },
       desc="A major mode for editing Java properties files.")
public class PropertiesMode extends GrammarCodeMode {

  public PropertiesMode (Env env) { super(env); }

  @Override public String langScope () {
    return "source.java-props";
  }

  @Override public Commenter commenter () {
    return new Commenter() {
      @Override public String docOpen () { return "##"; }
      @Override public String docPrefix () { return "##"; }
      // TODO: ! is also a comment start character, sigh...
      @Override public String linePrefix () { return "#"; }
    };
  }
}
